src/Features/CSharp/Portable/SignatureHelp/LightweightOverloadResolution.cs

```diff
@@ -0,0 +1,254 @@
+                specified.AsRef()[parameterIndex] = true;
+        }
+
+        for (var i = 0; i < specified.Count; i++)
+        {
+            if (!specified[i])
+                return i;
+        }
+
+        return 0;
+    }
+
+    /// <summary>
+    /// Find the parameter index corresponding to each argument provided
+    /// </summary>
+    private bool TryPrepareArgumentToParameterMap(IMethodSymbol method, ref TemporaryArray<int> argumentToParameterMap)
+    {
+        Contract.ThrowIfTrue(argumentToParameterMap.Count != arguments.Count);
+
+        var currentParameterIndex = 0;
+        var seenOutOfPositionArgument = false;
+        var inParams = false;
+
+        for (var argumentIndex = 0; argumentIndex < arguments.Count; argumentIndex++)
+        {
+            // Went past the number of parameters this method takes, and this is a non-params method.  There's no
+            // way this could ever match.
+            if (argumentIndex >= method.Parameters.Length && !inParams)
+                return false;
+
+            var argument = arguments[argumentIndex];
+            if (argument is { NameColon.Name.Identifier.ValueText: var argumentName })
+            {
+                // If this was a named argument but the method has no parameter with that name, tehre's definitely
+                // no match.  Note: this is C# only, so we don't need to worry about case matching.
+                var namedParameterIndex = method.Parameters.IndexOf(p => p.Name == argumentName);
+                if (namedParameterIndex < 0)
+                    return false;
+
+                if (namedParameterIndex != currentParameterIndex)
+                    seenOutOfPositionArgument = true;
+
+                AddArgumentToParameterMapping(argumentIndex, namedParameterIndex, ref argumentToParameterMap);
+            }
+            else if (IsEmptyArgument(argument.Expression))
+            {
+                // We count the empty argument as a used position
+                if (!seenOutOfPositionArgument)
+                    AddArgumentToParameterMapping(argumentIndex, currentParameterIndex, ref argumentToParameterMap);
+            }
+            else if (seenOutOfPositionArgument)
+            {
+                // Unnamed arguments are not allowed after an out-of-position argument
+                return false;
+            }
+            else
+            {
+                // Normal argument.
+                AddArgumentToParameterMapping(argumentIndex, currentParameterIndex, ref argumentToParameterMap);
+            }
+        }
+
+        return true;
+
+        void AddArgumentToParameterMapping(int argumentIndex, int parameterIndex, ref TemporaryArray<int> argumentToParameterMap)
+        {
+            Debug.Assert(parameterIndex >= 0);
+            Debug.Assert(parameterIndex < method.Parameters.Length);
+
+            inParams |= method.Parameters[parameterIndex].IsParams;
```

Comment:
should this be `isParams`?

Reply:
no, i think this means "user is currently *in* the params parameters).